storage/aptosdb/src/db/include/aptosdb_reader.rs

```diff
@@ -886,14 +886,19 @@ impl AptosDB {
     ) -> Result<TransactionWithProof> {
         self.error_if_ledger_pruned("Transaction", version)?;
 
-        let proof = self
+        let mut proof = self
             .ledger_db
             .transaction_info_db()
             .get_transaction_info_with_proof(
                 version,
                 ledger_version,
                 self.ledger_db.transaction_accumulator_db(),
             )?;
+
+        let aux_data_opt = self.get_transaction_auxiliary_data_by_version(version).ok();
+        proof.inject_auxiliary_error_data(aux_data_opt);
```

Comment:
hmm... this will make the transaction info invalid (breaks the hash).. 

please revert